flake-module.nix

```diff
@@ -159,9 +202,22 @@ in
                     hlint;
                 };
                 buildTools = lib.attrValues (defaultBuildTools hp // cfg.buildTools hp);
-                package = cfg.modifier (hp.callCabal2nixWithOptions cfg.name cfg.root "" { });
+                package =
+                  if cfg.topLevelPackages.enableMultiPkgs then
+                    hp.${cfg.topLevelPackages.defaultPkg}
+                  else
+                    cfg.modifier (hp.callCabal2nixWithOptions cfg.name cfg.root "" { });
```

Comment:
I think it's desirable to use the overlay (`haskellPackages.extend`) even in case of a single package, because it removes the possibility of accidentally using the Nixpkgs version of the package instead of the locally defined one.